src/nipple.js

```diff
@@ -300,13 +302,14 @@ Nipple.prototype.computeDirection = function (obj) {
     //      /   \
     //     /DOWN \
     //
-    if (rAngle > angle45 && rAngle < (angle45 * 3)) {
+    if (rAngle > angle45 && rAngle < (angle45 * 3) && !obj.horizontalOnly) {
         direction = 'up';
-    } else if (rAngle > -angle45 && rAngle <= angle45) {
+    } else if (rAngle > -angle45 && rAngle <= angle45 && !obj.verticalOnly) {
         direction = 'left';
-    } else if (rAngle > (-angle45 * 3) && rAngle <= -angle45) {
+    }else if (rAngle > (-angle45 * 3) && rAngle <= -angle45 &&
+    !obj.horizontalOnly) {
```

Comment:
formatting, add a tab after the line break and a space between the `}` and `else`